soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,215 @@
+		self.variable_square = 5.5
+
+		#NEVER CHANGE THIS
+		self.length = constants.Field.Length
+		self.width = constants.Field.Width
+
+		#if the square size is smaller than the width shrink it down otherwise
+		#max value is the width of the field.
+		self.variable_square = min(self.variable_square, self.width, self.length)
+
+		#radius of robot
+		self.bot_buffer = constants.Robot.Radius * 3
+		# the largest and smallest x and y postion possible for the square
+		self.max_x = (self.variable_square / 2 - self.bot_buffer)
+		self.min_x = -self.max_x
+		self.max_y = self.length/2 + self.max_x
+		self.min_y = self.length/2 - self.max_x
+
+		# the four courners
+		self.square_points = [
+			robocup.Point(self.max_x, self.max_y),
+			robocup.Point(self.min_x, self.max_y),
+			robocup.Point(self.min_x, self.min_y),
+			robocup.Point(self.max_x, self.min_y)]
+
+		# speed of the hunting robots
+		#TODO Create python pull from Config values. Currently this breaks the world.
+		#tmp = robocup.Configuration.FromRegisteredConfigurables().nameLookup("MotionConstraints/Max Velocity").value
+		self.normal_speed = 1.0 #tmp 
+		# speed of the defending robots can decrease value to make it easier for offense
+		self.defense_speed = self.normal_speed * 2 / 3 #self.normal_speed/2.0
+
+		# picks the direction to pass to. TODO make actual smart pass selection
+		self.direction = 1
+
+	# constanly running changes and updates
+	def execute_running(self):
+		# checks if there is an offensive behavior for the various plays and set the normal speed
+		for sub in self.subbehaviors_by_name() :
+			if (sub != 'passer') :
+				robo = self.subbehavior_with_name(sub).robot
+				# checks if there is a chasing robot and set their speed to defense speed
+				if (robo != None) :
+					if (sub == 'chasing'):
+						robo.set_max_speed(self.defense_speed)
+					# otherwise sets to the offensive normal speed rate
+					else:
+						robo.set_max_speed(self.normal_speed)		
+
+		# chasing robot positon should always follow the ball within a smaller inside box of the
+		# four corners.
+		if (self.has_subbehavior_with_name('chasing')) :
+			self.chaser.pos = self.cut_off_pos(main.ball().pos)
+
+		#draw the four courner field
+		# takes in the square points and form lines and create a square on the field
+		for i in range(len(self.square_points)) :
+			main.system_state().draw_line(robocup.Line(self.square_points[i], 
+				self.square_points[(i + 1) % 4]), (135, 0, 255), "Square")
+
+		# speed of the hunting robots
+		#TODO Create python pull from Config values. Currently this breaks the world.
+		#tmp = copy.deepcopy(robocup.Configuration.FromRegisteredConfigurables().nameLookup("MotionConstraints/Max Velocity").value)
+		self.normal_speed = 1.0 # tmp
+		#speed of the defending robots can decrease value to make it easier for offense
+		self.defense_speed = 2 * self.normal_speed/ 3.0 #self.normal_speed/2.0#self.variable_square/(min(self.width,self.length) * 2) * self.normal_speed
```

Comment:
You can take out the `self.variable_square...`

Reply:
Seconding Joe's comment